tests/test_text_replacement.py

```diff
@@ -0,0 +1,216 @@
+import unittest
+
+from kmk.keys import ALL_ALPHAS, ALL_NUMBERS, KC
+from kmk.modules.text_replacement import Character, Phrase, Rule, TextReplacement
+from tests.keyboard_test import KeyboardTest
+
+
+class TestTextReplacement(unittest.TestCase):
+    def setUp(self) -> None:
+        self.symbols = '`-=[]\\;\',./~!@#$%^&*()_+{}|:\"<>?'
+        self.everything = ALL_NUMBERS + ALL_ALPHAS + ALL_ALPHAS.lower() + self.symbols
+        self.test_dictionary = {'aa': 'b', 'b': 'aa', '!': '@', 'dccc': 'dcbb'}
+        self.text_replacement = TextReplacement(self.test_dictionary)
+        self.keyboard = KeyboardTest(
+            [self.text_replacement],
+            [
+                [KC.A, KC.B, KC.N1, KC.LSHIFT, KC.LCTRL, KC.C, KC.D],
+            ],
+            debug_enabled=True,
```

Comment:
```suggestion
            debug_enabled=False,
```
default for debugging is "off"